src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -794,9 +803,7 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
         master_template.add_secure_parameter('workspaceId', workspace_id)
         if os_type.lower() == 'linux':
             vm_mmaExtension_resource = build_vm_linux_log_analytics_workspace_agent(cmd, vm_name, location)
-            vm_daExtensionName_resource = build_vm_daExtension_resource(cmd, vm_name, location)
             master_template.add_resource(vm_mmaExtension_resource)
-            master_template.add_resource(vm_daExtensionName_resource)
         elif os_type.lower() == 'windows':
             vm_mmaExtension_resource = build_vm_windows_log_analytics_workspace_agent(cmd, vm_name, location)
             master_template.add_resource(vm_mmaExtension_resource)
```

Comment:
seem `Linux` and `windows` have the same logic for mmaExtension now. The if-elif is not necessary now.

Reply:
good catch.

Reply:
I double check the logic. Looks like we still need this one since we need to build different templates for different os. Meanwhile, I have another `else` logic for other os type except for `linux` and `windows`.